ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp

```diff
@@ -167,6 +167,17 @@ ttnn::Tensor ReshapeViewOperation::invoke(const ttnn::Tensor& tensor, const ttnn
     if (tensor_shape == shape) {
         return tensor;
     }
+
+    // Just edit shape if shape has a 0 dimension
+    if (tensor.volume() == 0) {
+        TT_ASSERT(shape.volume() == 0 && "tensor's volume is 0, but shape's volume is not 0");
+        TT_ASSERT((tensor.storage_type() != StorageType::MULTI_DEVICE &&
```

Comment:
Please use `TT_FATAL`

`TT_ASSERT` is only really hit in debug mode, and not run on CI by default.